### ISSUE TITLE
UnboundLocalError when trying to add a new slice for the first time.

### ISSUE THREAD

**Issue by rfilmyer** [airbnb/superset#1795]
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if any
- [X] I have reproduced the issue with at least the latest released version of superset (*at least the newest version from pip*)
- [X] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.14.1

### Expected results
I should see a page where I can add new slices.

### Actual results
A 500 error and this stacktrace:

```
2016-12-07 15:30:30,701:ERROR:superset:Exception on /slicemodelview/add [GET]
Traceback (most recent call last):
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/flask_appbuilder/security/decorators.py", line 26, in wraps
    return f(self, *args, **kwargs)
  File "/Users/roger/projects/superset/lib/python3.5/site-packages/superset/views.py", line 858, in add
    redirect_url = "/r/msg/?url={}&msg={}".format(url, msg)
```

which points to [this line of code](https://github.com/airbnb/superset/blob/0.14.1/superset/views.py#L858).

### Steps to reproduce
* Go to the "slices" tab.
* Click the `+` (Add a new record) button
* 500 error.

I have basically just installed superset from scratch. I only have one database, a Redshift (via postgres/psycopg2) database where I can see tables and columns in the SQL lab.

**Comment by xrmx** [comment]
Indeed, code looks broken, thanks for reporting.

**Comment by bkyryliuk** [comment]
hm, broken on master :(

**Comment by pgp44** [comment]
Is there a workaround for this?

Tx

Peter

**Comment by xrmx** [comment]
Dup of #2564, will be fixed by #2565
